dlog/kimchi/src/alphas.rs

```diff
@@ -4,6 +4,10 @@
 //! It is important that different constraints use different powers of alpha,
 //! as otherwise they can interact and potentially cancel one another.
 //! (The proof is in the use of the Schwartz-Zipple lemma.)
+//! As such, we want two properties from this:
+//!
+//! - we should keep track of a mapping between type of constraint and range of powers
+//! - when powers of alphas are used, we should ensure that no more no less are used
```

Comment:
I like your way of engineering this safety check in.

For unit tests are there even more exhaustive checks we could do?

Reply:
it'd be nice to use proptests to test different configurations in tests

Reply:
👍🏼